### ISSUE TITLE
Does it work with `@Controller` in nestjs?

### ISSUE THREAD

**Issue by CatsMiaow** [Papooch/decorate-all#2]
https://github.com/CatsMiaow/typescript-starter/blob/master/src/app.controller.ts#L7
In the sample code above, `test2` outputs normally, but if you add (uncomment) `@DecorateAll`, the controller's route outputs a 404 page.

https://github.com/Papooch/decorate-all/blob/main/src/lib/decorate-all.decorator.ts#L35
If remove `Object.defineProperty` it doesn't throw a 404, but the decorator doesn't work.

**Comment by Papooch** [comment]
Since you're replacing the entire function definition as opposed to only adding metadata (which was the original intended purpose), any other previously defined metadata (those defined by the `@Get`) are lost, because class decorators are applied after methdo decorators.

You can work around it by explicitly copying the metadata in your custom decorator using this helper function:

```ts
/**
 * Copies all metadata from one object to another.
 * Useful for overwriting function definition in
 * decorators while keeping all previously
 * attached metadata
 *
 * @param from object to copy metadata from
 * @param to object to copy metadata to
 */
export function copyMetadata(from: any, to: any) {
    const metadataKeys = Reflect.getMetadataKeys(from);
    metadataKeys.map((key) => {
        const value = Reflect.getMetadata(key, from);
        Reflect.defineMetadata(key, value, to);
    });
}
```
and in your `debug-log.decorator.ts`:

```ts

const newMethod = async function (...args: unknown[]): Promise<unknown> {
      const result: unknown = await originalMethod.apply(this, args);

      console.log(context, Date.now());
      return result;
};
copyMetadata(originalMethod, newMethod);
descriptor.value = newMethod;
```

This is clearly an oversight on my end and I'll probably incorporate it into the package itself.

EDIT: you also need to install the [`reflect-metadata`](https://www.npmjs.com/package/reflect-metadata) package

**Comment by Papooch** [comment]
It is now fixed in `v1.1.0` and published to npm.
